### ISSUE TITLE
Import/export CAR DAG archives

### ISSUE THREAD

**Issue by lidel** [ipfs/ipfs-webui#1798]
Given that:
- `ipfs dag import|export` exists
- go-ipfs 0.9+ supports exporting DAG archive via any public gateway (`/api/v0/dag/export`)
- tools like [ipfs-car](https://www.npmjs.com/package/ipfs-car) and [ipfs-content-addressed-archiver](https://github.com/olizilla/ipfs-content-addressed-archiver) make it easier for people to produce/consume content-addressed archives

We should support "DAG import/export"  in ipfs-webui somehow.

## Import DAG

Import could  work the same as "From IPFS" :

> ![2021-06-01--13-01-05](https://user-images.githubusercontent.com/157609/120312856-8a5d6380-c2d9-11eb-85a4-a6e602c93020.png)

We most likely want to add an explicit "From CAR Archive"  item here  (personally, I like the idea of a whimsical car icon  :)), but "stroke data" from [ipfs-css](https://github.com/ipfs-shipyard/ipfs-css#icons) would also do.

## Export DAG

Add complimentary "Export DAG" to context menu on Files screen:

> ![2021-06-11--16-08-29](https://user-images.githubusercontent.com/157609/121699752-7ed71d00-cacf-11eb-873e-6b73b1ef0abd.png)

**Comment by Gozala** [comment]
Seems like it would be a good idea to do it after https://github.com/ipfs/js-ipfs/issues/2745

**Comment by hacdias** [comment]
@lidel as I mentioned on https://github.com/ipfs/ipfs-webui/pull/1838#issuecomment-901733616, we'll need to update to the latest HTTP client across WebUI and the Explore page to support the import part. The export is done on #1837. I don't think I should work on the upgrade right now as that would take a long time working only 10h a week. Updating the http client will also require updating everything else connected: multiformats, ipld, multicodecs, etc.

I think we could merge the export button and evaluate how much effort is needed for the import part.

**Comment by hacdias** [comment]
@lidel I agree with your worry. I didn't think about that, but it's a good point.

**Comment by lidel** [comment]
Reopening since we also need https://github.com/ipfs/ipfs-webui/pull/1838

**Comment by hsn10** [comment]
Add maximum CAR size so output can be split into multiple files. Standard between services seems to accept 100 MB CAR files for uploads.

**Comment by lidel** [comment]
AS of 2025Q1 we do have support for CAR export:

> ![2025-01-13_20-29](https://github.com/user-attachments/assets/1300161b-afb4-4188-bf45-614892520103)



Remaining work here is to add support for CAR Import in the `Import` menu:

> ![image](https://github.com/user-attachments/assets/dd27207e-f456-4a2a-a083-724a6af7d5b2)


To wired it up:
1. Add "From CAR" button to Import menu.
1. Clicking "From CAR" should open dialog similar to one behind "From IPFS", but 
   - instead of field where one puts path to import, have button that triggers file picker 
   - name field allow adjusting the name that will be assigned to the root CID
    - use `ipfs.dag.import` from https://www.npmjs.com/package/kubo-rpc-client, and then `ipfs.files.cp /ipfs/root-cid-from-dag-import /path-or-name-in-mfs`
